### PR TITLE
sql: don't panic on unexpected null in kvfetcher

### LINKED ISSUE #35083: sql: don't panic on unexpected null in kvfetcher
CockroachDB currently panics if a NULL is detected on disk for a non-nullable column. This is pretty much an overreaction to a situation that can arise due to bugs in our code. We should return an error in the query instead.

### PR DESCRIPTION
**Pull request by asubiotto**
Closes #35083

Release note: None

### CODE REVIEW

File: pkg/sql/row/fetcher.go [hunk 1]
```diff
@@ -1249,16 +1249,15 @@ func (rf *Fetcher) finalizeRow() error {
...
 					}
 				}
-				if rf.isCheck {
-					return scrub.WrapError(scrub.UnexpectedNullValueError, errors.Errorf(
-						"Non-nullable column \"%s:%s\" with no value! Index scanned was %q with the index key columns (%s) and the values (%s)",
-						table.desc.Name, table.cols[i].Name, table.index.Name,
-						strings.Join(table.index.ColumnNames, ","), strings.Join(indexColValues, ",")))
-				}
-				panic(fmt.Sprintf(
+				err := errors.Errorf(
```
Comment (jordanlewis) [inline pkg/sql/row/fetcher.go]: Make this a pgerror AssertionError.

Comment (asubiotto) [reply pkg/sql/row/fetcher.go]: Done.

**Review by jordanlewis — APPROVED** [review]
Thanks! Please backport as well.

**Comment by asubiotto** [comment]
bors r=jordanlewis

**Pull request merged**